_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,273 @@
+    context resource object.
+  - `successor-version`: links to the version which immediately succeeded the
+    context resource object.
+  - `prior-working-copy`: links to the working copy which immediately preceded
+    the context resource object.
+  - `subsequent-working-copy`: links to the working copy which immediately
+    preceded the context resource object.
+
+A server **MAY** provide an array of any of these links to support branching.
+
+For example, in the following version history:
+
+```
+    _c_
+   /   \
+  /__b__d__   __f__   __h
+ /         \ /     \ /
+a           e       g
+```
+
+`g` is the latest version. Both `a` and `e` were previously the latest version.
+No other revisions were ever the latest version. In this example, the following
+links could be provided:
+
+| Revision | `latest-version` | `working-copy` | `predecessor-version` | `successor-version` | `prior-working-copy` | `subsequent-working-copy` |
+| :------: | :--------------: | :------------: | :-------------------: | :-----------------: | :------------------: | :-----------------------: |
+| `a`      | `g`              | `h`            | no link               | `e`                 | no link              | `b`, `c`                  |
+| `b`      | `g`              | `h`            | `a`                   | `e`                 | `a`                  | `c`                       |
+| `c`      | `g`              | `h`            | `a`                   | `e`                 | `b`                  | `d`                       |
+| `d`      | `g`              | `h`            | `a`                   | `e`                 | `b`, `c`             | `e`                       |
+| `e`      | `g`              | `h`            | `a`                   | `g`                 | `d`                  | `f`                       |
+| `f`      | `g`              | `h`            | `e`                   | `g`                 | `e`                  | `g`                       |
+| `g`      | no link          | `h`            | `e`                   | no link             | `f`                  | `h`                       |
+| `h`      | `g`              | no link        | `g`                   | no link             | `g`                  | no link                   |
+
+> Note: In this example, `f` has both a `predecessor-version` and
+> `prior-working-copy` link to `e` because `e` was a version and `e` was also the
+> revision to which changes could be applied prior to `f`'s creation.
+
+# Version History
+
+A server **MAY** provide a "version history" endpoint. The primary data
+of a response document from a version history endpoint must be a collection of
+resource objects.
+
+Unless an `id` contains version information, the `type` and `id` members of each
+resource object in the collection **MUST** be the same.
+
+If provided, resource objects' `self` links **MUST NOT** be the same.
+
+# Version Negotiators
+
+## ID-Based Version Negotiator
```

Comment:
MUST this be implemented in every server supporting versioning?

Reply:
No, a server that versions by updated timestamp might use a `timestamp` negotiator as it's "most specific" negotiator. It might not have revision IDs.

Reply:
For example, a server could have a negotiator named `at` as long as it doesn't support branched versioning or forward revisions. That would allow a client to request any timestamp and get the appropriate revision for that time period. The version history would just give `self` links equal to `at:{updated timestamp}`.

Reply:
Then we probably want language to clarify this section and say that these are optional version negotiators. Also worth noting that an implementor should have at least one version negotiator defined.

Reply:
f0d73db82a26d8ab999b38a582969e112b864351

Reply:
I did not mention that they should have one defined. That seems unnecessary as the specification can't work without it. No?

Reply:
See: a4329d5